src/Models.hs

```diff
@@ -0,0 +1,23 @@
+{-# LANGUAGE DeriveAnyClass     #-}
+{-# LANGUAGE DeriveGeneric      #-}
+{-# LANGUAGE DerivingStrategies #-}
+
+module Models
+  ( User(..)
+  ) where
+
+import           Data.Aeson                 as JSON
+import           Database.PostgreSQL.Simple (FromRow, ToRow)
+import           GHC.Generics
+
+data User = User
+  { userId   :: Int
+  , userName :: String
+  , userage  :: Maybe Int
+  } deriving (Generic, Eq, Show)
+    deriving anyclass (ToRow, FromRow)
+
+instance ToJSON User where
+  toEncoding = genericToEncoding defaultOptions
+
+instance FromJSON User
```

Comment:
By the way, with `anyclass` deriving you can also derive generic instances for `ToJSON` and `FromJSON` (see [this](https://kowainik.github.io/posts/deriving#generic-anyclass)).

Which, incidentally, is what you did for `FromJSON User`! (I just learned this thanks to you haha). See the [docs on deriving anyclass](https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/derive_any_class.html):

![image](https://github.com/carlota-moh/servant-blog/assets/22799724/d62c5036-ad1f-46ee-b96c-afa13301b727)

So you can simply remove the manual instances here and add `ToJSON` and `FromJSON` to the `deriving anyclass` list you have.

Reply:
Yeah I think that is legacy and I simply forgot to update it, thnx!!